api/server/cluster_test.go

```diff
@@ -0,0 +1,53 @@
+package server
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/libopenstorage/openstorage/api"
+	client "github.com/libopenstorage/openstorage/api/client/cluster"
+	"github.com/libopenstorage/openstorage/cluster"
+	mockcluster "github.com/libopenstorage/openstorage/cluster/mock"
+	"github.com/stretchr/testify/assert"
+
+	"github.com/golang/mock/gomock"
+	"github.com/kubernetes-csi/csi-test/utils"
+)
+
+func TestServerNodeStatus(t *testing.T) {
+
+	// Save already set value of cluster.Inst to set it back
+	// when we finish the tests by the defer()
+	oldInst := cluster.Inst
```

Comment:
If running in a test context, why would you need this? You can always override the cluster.Inst function definition.

Reply:
We need this, because you should always place tests back to a clean state before running the the next test. The next one may want to use the real Cluster.